src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,123 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * Represents the text parser for Duke. When parsing a user input, the parser
+ * also handles the updating of the associated TaskList.
+ *
+ * @author Joshua Yong
+ */
+public class Parser {
+
+    // The TaskList which the Parser updates
+    private TaskList tasks;
+
+    /**
+     * Class constructor.
+     *
+     * @param tasks The TaskList to be updated during parsing.
+     */
+    public Parser(TaskList tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Parses the user input String, updating the associated TaskList.
+     * Also returns an appropriate String to be displayed to the user.
+     *
+     * @param userInput The String inputted by the user.
+     * @return A String to be displayed to the user.
+     * @throws DukeException if the user input is invalid.
```

Comment:
Perhaps you could capitalise the first letter of the sentence after the word DukeException?

```suggestion
     * @throws DukeException If the user input is invalid.
```